### PR TITLE
various: tidy up S3 URLs

### PR DESCRIPTION
**Pull request by vszakats**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making any changes to a cask, existing or new, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [ ] `brew audit --cask --online <cask>` is error-free.
- [ ] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [ ] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [ ] Checked the cask was not [already refused](https://github.com/search?q=repo%3AHomebrew%2Fhomebrew-cask+is%3Aclosed+is%3Aunmerged+&type=pullrequests) (add your cask's name to the end of the search field).
- [ ] Checked the cask is submitted to [the correct repo](https://docs.brew.sh/Acceptable-Casks#finding-a-home-for-your-cask).
- [ ] `brew audit --cask --new <cask>` worked successfully.
- [ ] `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --cask <cask>` worked successfully.
- [ ] `brew uninstall --cask <cask>` worked successfully.

---

Prefer:
- `<prefix>.s3.amazonaws.com` (modern) URL flavour over `s3.amazonaws.com/<prefix>`.
  (this is always supported by S3 for dotless, lowercase prefixes.)
- region-agnostic URLs unless they do a permanent redirect to region-specific.
- `s3-<region>.amazonaws.com` over `s3.<region>.amazonaws.com`.

All hand checked.

### CODE REVIEW

**Comment by vszakats** [comment]
All failures seem unrelated to these updates:
- unmodified homepage unreachable: `arctype`
- packages needing version updates: `datadog-agent`, `ripx`, `rstudio@daily`
- livecheck check telling version doesn't match `''`?: `data-integration`
- broken check for a working livecheck URL: `izotope-product-portal`
  (the URL works from here from a browser)

**Comment by bevanjkay** [comment]
I'm not sure how much it matters either way, but I would suggest that most of these instances were consistent with what is broadcast by the vendor, whether it be linked from their homepage, appcast or other medium. 

I don't feel strongly whether we should stick to a standard, or mirror what is used upstream.

**Comment by SMillerDev** [comment]
I think requiring changes here will complicate contributions. I can't imagine it'll be easy if you want to make a cask, copy an upstream download link and `brew style` tells you it's wrong.

**Comment by vszakats** [comment]
The prefixed URLs are the ones recommened by Amazon (for a long time now). I've been doing similar updates for a few years way back when. These URLs also play better with corporate (or non-corporate) firewalls. Region-agnostic URLs most likely play better (faster) when downloading from a different region, and one of the points of S3.

I'm not sure how this complicates controbutions. There is no enforcement for this AFAIK.

**Comment by vszakats** [comment]
I don't think it's realistic to expect me to push through an brew audit update for this. I haven't done it for a long time and don't have the guts or time to touch it. Furthermore the regsion-agnostic URLs almost always need manual checks, because S3 _can_ be configured to play bad with them, and in a few rare cases, they do.

I haven't touched this in the last 5 years and 50-ish fallouts that accumulated throughout this time isn't that much to justify a brew audit effort IMO.

**Comment by vszakats** [comment]
`data-integration` now fails because the homepage is apparently temporarly down.

version bumped `datadog-agent`, `rstudio@daily`.

will delete commits for the 3 broken casks.

**Comment by vszakats** [comment]
The remaining issues are still unrelated to this PR, and they also seem to be detection issues:
- `data-integration`: homepage is temporarly down, it was still up on the first CI run.
- `izotope-product-portal`: livecheck URL is actually working, and the returned version is 1.4.8, which is already offered by the cask.

**Comment by vszakats** [comment]
drop for pre-existing issues:
anythingllm
data-integration
itubedownloader
izotope-product-portal
polypad

File: Casks/m/movist-pro.rb [hunk 1]
```diff
@@ -2,14 +2,14 @@
   version "2.11.4"
   sha256 "ed5110e1b7b215febbf34bf192158b7670d2c07d7140cad10a821d0f7232e8f6"
 
-  url "https://s3.ap-northeast-2.amazonaws.com/update.cocoable.com/MovistPro_#{version}.dmg",
-      verified: "s3.ap-northeast-2.amazonaws.com/update.cocoable.com/"
+  url "https://s3-ap-northeast-2.amazonaws.com/update.cocoable.com/MovistPro_#{version}.dmg",
```
Comment (daeho-ro) [inline Casks/m/movist-pro.rb:5]: I think this url should be `https://update.cocoable.com.s3.amazonaws.com/` same as the others.

Comment (vszakats) [reply Casks/m/movist-pro.rb:5]: You are right, thanks for catching it!

**Comment by vszakats** [comment]
Sigh, failure because one cask received an upstream update.

**Comment by krehel** [comment]
If no one has objections to this, I'm fine for ✅, but do want to solve some automated way of identifying these going forward.

**Comment by vszakats** [comment]
A rough sketch for the multi-step URL transformation:
1. `s3-accelerate.` → `s3.`: strip "accelerate"
1. `s3.([a-z0-9-]+).amazonaws.com` → `s3-\1.amazonaws.com`: prefer single-level
1. `s3(-[a-z0-9-]+)?.amazonaws.com/([a-z][a-z0-9_-]*)/` → `\2.s3\1.amazonaws.com/`: prefer prefix-style
1. `.s3(-[a-z0-9-]+).amazonaws.com/` → `.s3.amazonaws.com/`: go region-agnostic — if returning 200. Or 301, which works if following the redirect, but I didn't go for it.

Double-check `-gov-` regions. (`elasticwolf` uses it here, and both download and homepage seem to be gone.)

**Review by razvanazamfirei — APPROVED** [review]

**Comment by razvanazamfirei** [comment]
This looks great! 👍 Thanks so much for your contribution! Without contributions like yours, it'd be impossible to keep our project going.

**Pull request merged**